### DISCUSSION TITLE
Make the generator extensible to allow decorators to be added

### DISCUSSION THREAD

**Discussion by carlfyffe-pp** [mikro-orm/mikro-orm#4046] [Ideas]
We use GraphQL and it would be great if we could inject GraphQL types during the generation phase. They align very closely with the entities so all we really need to do is be able to map to entity and property generation. Having the generator call out to any plugins that have been added would be tremendous here. This would allow any of the GraphQL libraries to plugin to your generation process. I feel like this could be used by many other types of libraries that depend on entities, not just GraphQL.

**Comment by B4nan** [comment]
I don't mind adding any kind of extension points, PR welcome for that, the entity generator part is quite simple and isolated package so should be easy to navigate.
